python_src/main.py

```diff
@@ -0,0 +1,52 @@
+import click
+import json
+from visualization import VehicleTrajectoryVisualizer
+from models import SimulationFactory, Simulation, categorize_report
+from models.ac3rp import CrashScenario
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.argument('scenario_file', type=click.Path(exists=True))
+def plot_ac3r(scenario_file):
+    """Take a JSON scenario file and plot the AC3R data points."""
+    VehicleTrajectoryVisualizer.plot_ac3r(scenario_file)
+
+
+@cli.command()
+@click.argument('scenario_file', type=click.Path(exists=True))
+def plot_ac3rp(scenario_file):
+    """Take a JSON scenario file and plot the AC3RPlus data points."""
+    VehicleTrajectoryVisualizer.plot_ac3rp(scenario_file)
+
+
+@cli.command()
+@click.argument('scenario_file', type=click.Path(exists=True))
+def run_from_scenario(scenario_file):
+    """Take a JSON scenario file and run the entire search algorithm."""
+    with open(scenario_file) as file:
```

Comment:
This is conceptually Ok, but we probably need to specify additional arguments to this command, for example, what fitness function to use, or where we can read some configurations (like mutation operators, speed min/max, and other parameters). For the moment I suggest we add a `#TODO` comment to remind us that we'll need more arguments later.

Reply:
Yeah, that is what we are planning to do. The current method is actually a simple method of taking data from JSON file with `cli` and using a new approach to execute it on BeamNG simulation. In term of other configurations, I will add a new method to provide that ability, but for now, the `#TODO` comment is enough for merging this big PR.